src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -819,5 +830,29 @@ private static ILogger GetLogger(HttpContext httpContext)
                 return loggerFactory.CreateLogger(typeof(RequestDelegateFactory));
             }
         }
+
+        private static void EnsureRequestTaskOfNotNull<T>(Task<T?> task)
+        {
+            if (task is null)
+            {
+                throw new InvalidOperationException("Task Result should not be null");
```

Comment:
Let's try to reference the exact type in the exception where we can. E.g. `Task<IResult>`, `ValueTask<string>`, etc... This also makes it clearer for future reader of the code that this check doesn't apply to all `Task<T>`.

Reply:
Ok 👍